### ISSUE TITLE
New rules: no-universal-selector, no-attribute-selector & no-combinators

### ISSUE THREAD

**Issue by bgriffith** [sasstools/sass-lint#694]
Got a few basic new rules I'd like to add to Sass-lint and I think all would probably be disabled by default a la `no-extends`. Be great to get your thoughts @sasstools/sass-lint-contributors esp. on `...-selector` vs `...-selectors`

- `no-universal-selector` 
- `no-attribute-selector`
- `no-combinators`

If it's all good, I'll push up the PR 🎉 

====

## No Universal Selector

Rule `no-universal-selector` will enforce that the `*` (universal) selector is not allowed to be used.

### Examples

When enabled, the following are disallowed:

```scss
* {
  content: 'foo';
}

* [lang^=en] {
  content: 'bar';
}

*.warning {
  content: 'baz';
}

*#maincontent {
  content: 'qux';
}

*:before,
*:after {
  content: 'norf';
}
```

===

## No Attribute Selector

Rule `no-attribute-selector` will enforce that the attribute selector is not allowed to be used.

### Examples

When enabled, the following are disallowed:

```scss
[autoplay] {
  content: 'foo';
}

[lang=en] {
  content: 'bar';
}

[lang~=en-us] {
  content: 'baz';
}

[lang|=us] {
  content: 'qux';
}

[href^="#"] {
  content: 'norf';
}

[href$=".com"] {
  content: 'foo';
}

[href*=news] {
  content: 'bar';
}
```

===

## No Combinators

Rule `no-combinators` will enforce that the combinators are not allowed to be used.

### Examples

When enabled, the following are disallowed:

```scss
.foo > .bar {
  content: 'foo';
}

.foo ~ .bar {
  content: 'bar';
}

.foo + .bar {
  content: 'baz';
}

.foo .bar {
  content: 'qux';
}
```

**Comment by DanPurdy** [comment]
I would make them plural in name to follow what we already do. i.e. no-meregeable-selectors, no-color-keywords but apart from that I think they're ok, i'll have a closer look though. I would have them disabled in our config by default though maybe?

With no-combinators have you checked with interpolated values

**Comment by bgriffith** [comment]
Ok plural it is! I also agree with disabling them by default (like `no-extends`).

I think I've got it covered but will add some more tests just to be safe.

**Comment by DanPurdy** [comment]
These will all be available with sass-lint 1.8.0
